plugins/automation/app/models/discourse_automation/stat.rb

```diff
@@ -0,0 +1,128 @@
+# frozen_string_literal: true
+#
+module DiscourseAutomation
+  class Stat < ActiveRecord::Base
+    self.table_name = "discourse_automation_stats"
+
+    def self.log(automation_id, run_time = nil)
+      if block_given? && run_time.nil?
+        start_time = Process.clock_gettime(Process::CLOCK_MONOTONIC)
+        result = yield
+        run_time = Process.clock_gettime(Process::CLOCK_MONOTONIC) - start_time
+        result
+      end
+    ensure
+      update_stats(automation_id, run_time)
+    end
+
+    def self.fetch_period_summaries
+      today = Date.current
+
+      # Define our time periods
+      periods = {
+        last_day: {
+          start_date: today - 1.day,
+          end_date: today,
+        },
+        last_week: {
+          start_date: today - 7.days,
+          end_date: today,
+        },
+        last_month: {
+          start_date: today - 30.days,
+          end_date: today,
+        },
+      }
+
+      result = {}
+
+      periods.each do |period_name, date_range|
+        builder = DB.build <<~SQL
+          SELECT
+            automation_id,
+            SUM(total_runs) AS total_runs,
+            SUM(total_time) AS total_time,
+            CASE WHEN SUM(total_runs) > 0
+              THEN SUM(total_time) / SUM(total_runs)
+              ELSE 0
+            END AS average_run_time,
+            MIN(min_run_time) AS min_run_time,
+            MAX(max_run_time) AS max_run_time
+          FROM discourse_automation_stats
+          WHERE date >= :start_date AND date <= :end_date
+          GROUP BY automation_id
+        SQL
+
+        stats = builder.query(start_date: date_range[:start_date], end_date: date_range[:end_date])
+
+        last_run_stats = DB.query_array <<~SQL
+          SELECT
+            automation_id,
+            MAX(last_run_at) AS last_run_at
+          FROM discourse_automation_stats
+          GROUP BY automation_id
+        SQL
+        last_run_stats = Hash[*last_run_stats.flatten]
+
+        stats.each do |stat|
+          automation_id = stat.automation_id
+          result[automation_id] ||= {}
+          result[automation_id][:last_run_at] = last_run_stats[automation_id]
+          result[automation_id][period_name] = {
+            total_runs: stat.total_runs,
+            total_time: stat.total_time,
+            average_run_time: stat.average_run_time,
+            min_run_time: stat.min_run_time,
+            max_run_time: stat.max_run_time,
+          }
```

Comment:
If all we want to do is assign the whole result, can we use `#query_hash` and assign it directly instead of having to dig out every column?

Reply:
not sure what you mean here?